lib/ldp.js

```diff
@@ -156,42 +154,40 @@ class LDP {
       return callback(error(500, "Can't parse container"))
     }
 
-    async.waterfall(
-      [
-        // add container stats
-        function (next) {
-          ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph, next)
-        },
-        // reading directory
-        function (next) {
-          ldpContainer.readdir(filename, next)
-        },
-        // Iterate through all the files
-        function (files, next) {
-          async.each(
-            files,
-            function (file, cb) {
-              let fileUri = url.resolve(reqUri, encodeURIComponent(file))
-              ldpContainer.addFile(ldp, resourceGraph, reqUri, fileUri, uri,
-                filename, file, cb)
-            },
-            next)
-        }
-      ],
-      function (err, data) {
+    // add container stats
+    new Promise((resolve, reject) =>
+      ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph,
+        err => err ? reject(err) : resolve())
```

Comment:
It's definitely a matter of taste but this is how I generally like things: when breaking function arguments up into several lines, I like to keep the closing paren `)` on its own line, indented to match the start of the line of the function call.

For example:

```js
ldpContainer.addContainerStats(
  ldp, reqUri, filename, resourceGraph,
  err => err ? reject(err) : resolve()
)
```

I also see this a lot when using callback functions - breaking the line after the arguments to the callback:
```js
ldpContainer.addContainerStats(ldp, reqUri, filename, resourceGraph, err =>
  err ? reject(err) : resolve()
)
```